lib/swagger_yard/property.rb

```diff
@@ -60,11 +60,29 @@ def initialize(name, types, description, options)
       @name, @description = name, description
       @required = options.include?('required')
       @nullable = options.include?('nullable')
+      @extensions = parse_extensions(options.select { |option| option.start_with?("x-") })
       @type = Type.from_type_list(types)
     end
 
-     def required?
+    def required?
       @required
     end
+
+    def internal?
+      extensions["x-internal"] == 'true'
```

Comment:
Do we want to downcase the input?  We could also go further like rails and support a bunch of different options: https://blog.eq8.eu/til/convert-string-true-and-string-false-to-boolean-with-rails.html, but maybe that is overkill

Reply:
Yes, I think that's a good idea. Especially if we start using this for other extensions. I'm going to defer that to another PR  though.